### PR TITLE
DOC: Add transformation grid documentation page

### PR DESCRIPTION
**Pull request by snowman2**

### CODE REVIEW

File: docs/transformation_grids.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
+.. _transformation_grids:
+
+Transformation Grids
+=====================
+
+Transformation grids are necessary when you are performing datum transformations.
```
Comment (jorisvandenbossche) [inline docs/transformation_grids.rst:6]: They are not "necessary" (transformations also work if the grids are not available?), but make the transformation more accurate?

File: docs/transformation_grids.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+What grids to download?
+-----------------------
+
+- Only using the :obj:`pyproj.crs.CRS` or :obj:`pyproj.Geod` classes? Then no grids are needed.
+
+- Have a machine that can hold and extra 500 MB - 1 GB of data? Then downloading all grids shouldn't be an issue.
+
+- Have a machine with limited space, a great network connection, and PROJ 7+? Look into `PROJ network <https://proj.org/usage/network.html>`__ capabilities.
+
```
Comment (jorisvandenbossche) [inline docs/transformation_grids.rst:72]: With PROJ 7, it will also download the subset of grids you need (and cache those) automatically?

Comment (snowman2) [reply docs/transformation_grids.rst:72]: By default it isn't enabled. You need to enable it. However, if you installed it via conda-forge it is enabled by default.

File: docs/transformation_grids.rst (hunk 1 above)
Comment (snowman2) [reply docs/transformation_grids.rst:6]: That is true. I guess it depends on your definition of necessary.

**Pull request merged**